Source/Core/Core/PowerPC/Jit64/RegCache/CachedReg.h

```diff
@@ -106,6 +123,7 @@ class PPCCachedReg
   Gen::OpArg default_location{};
   Gen::OpArg location{};
   bool away = false;  // value not in source register
+  bool discarded = false;
```

Comment:
Since a discarded vallue can never have a valid location, perhaps consider implementing this as `location == std::nullopt` for a `location` of type `std::optional<Gen::OpArg>`.

Reply:
Done.